source/plugin/data/custom/datamanipulators.rst

```diff
@@ -144,6 +149,45 @@ change is made to the format of your serialized data, and use :ref:`content-upda
         return container;
     }
 
+Registration
+============
+
+Registering your ``DataManipulator`` allows it to be accessible by Sponge and by other plugins in a generic way. The
+server/plugin can create copies of your data and serialize/deserialize your data without referencing any of your classes
+directly.
+
+To register a ``DataManipulator`` Sponge has the :javadoc:`DataRegistration#builder()` helper. This will build a
+:javadoc:`DataRegistration` and automatically register it with the server.
+
+.. code-block:: java
+
+    import org.spongepowered.api.event.game.state.GameInitializationEvent;
+    import org.spongepowered.api.data.DataRegistration;
+
+    import org.example.MyCustomData;
+    import org.example.ImmutableCustomData;
+    import org.example.CustomDataBuilder;
+
+    @Listener
+    public void onInit(GameInitializationEvent event) {
+      DataRegistration.<MyCustomData, ImmutableCustomData>builder()
+          .setDataClass(MyCustomData.class)
+          .setImmutableDataClass(ImmutableCustomData.class)
+          .setBuilder(new CustomDataBuilder())
+          .setManipulatorId("my_custom")
+          .buildAndRegister(this);
+    }
+
+Note that in the above we perform this action during initialization. If you register your data too late in the lifecycle
+of the server, your ``DataManipulator`` will not be available for the server to deserialize if required.
+
+.. warning::
+
+    Any data that was serialized prior to ``6.0.0`` will *not* be recognised when registering this way. To ensure that
+    legacy data serializations are deserialized, you should call
+    :javadoc:`DataManager#registerLegacyManipulatorIds(String, DataRegistration<?, ?>)`.
+    The ID is the class' canonical name, such as ``com.example.MyCustomData``.
```

Comment:
This is also utilized in the event that legacy data is being "upgraded" from an older manipulator id.